elasticsearch-model/spec/elasticsearch/model/naming_inheritance_spec.rb

```diff
@@ -19,105 +19,185 @@
-      class Dog < ::Animal
-      end
+    around(:all) do |example|
+      original_value = Elasticsearch::Model.settings[:inheritance_enabled]
+      Elasticsearch::Model.settings[:inheritance_enabled] = true
+      example.run
+      Elasticsearch::Model.settings[:inheritance_enabled] = original_value
     end
 
-    class ::Cat < ::Animal
-      extend ActiveModel::Naming
 
-      extend  Elasticsearch::Model::Naming::ClassMethods
-      include Elasticsearch::Model::Naming::InstanceMethods
+    describe '#index_name' do
+
+      it 'returns the default index name' do
+        expect(TestBase.index_name).to eq('test_bases')
+        expect(TestBase.new.index_name).to eq('test_bases')
+      end
 
-      index_name "cats"
-      document_type "cat"
+      it 'returns the explicit index name' do
+        expect(Animal.index_name).to eq('mammals')
+        expect(Animal.new.index_name).to eq('mammals')
+
+        expect(Cat.index_name).to eq('cats')
+        expect(Cat.new.index_name).to eq('cats')
+      end
+
+      it 'returns the ancestor index name' do
+        expect(Dog.index_name).to eq('mammals')
+        expect(Dog.new.index_name).to eq('mammals')
+      end
+
+      it 'returns the ancestor index name for namespaced models' do
+        expect(::MyNamespace::Dog.index_name).to eq('mammals')
+        expect(::MyNamespace::Dog.new.index_name).to eq('mammals')
+      end
     end
 
-  end
+    describe '#document_type' do
 
-  after(:all) do
-    remove_classes(TestBase, Animal, MyNamespace, Cat)
-  end
+      it 'returns nil' do
+        expect(TestBase.document_type).to be_nil
+        expect(TestBase.new.document_type).to be_nil
+      end
 
-  around(:all) do |example|
-    original_value = Elasticsearch::Model.settings[:inheritance_enabled]
-    Elasticsearch::Model.settings[:inheritance_enabled] = true
-    example.run
-    Elasticsearch::Model.settings[:inheritance_enabled] = original_value
-  end
+      it 'returns the explicit document type' do
+        expect(Animal.document_type).to eq('mammal')
+        expect(Animal.new.document_type).to eq('mammal')
 
+        expect(Cat.document_type).to eq('cat')
+        expect(Cat.new.document_type).to eq('cat')
+      end
 
-  describe '#index_name' do
+      it 'returns the ancestor document type' do
+        expect(Dog.document_type).to eq('mammal')
+        expect(Dog.new.document_type).to eq('mammal')
+      end
 
-    it 'returns the default index name' do
-      expect(TestBase.index_name).to eq('test_bases')
-      expect(TestBase.new.index_name).to eq('test_bases')
+      it 'returns the ancestor document type for namespaced models' do
+        expect(::MyNamespace::Dog.document_type).to eq('mammal')
+        expect(::MyNamespace::Dog.new.document_type).to eq('mammal')
+      end
     end
+  end
+
+  context 'when using proxy' do
+    before(:all) do
+      TestBase = Class.new do
+        extend ActiveModel::Naming
+
+        include Elasticsearch::Model
```

Comment:
I could just copy the required code from `Elasticsearch::Model` in here but that would result in tests passing even if the `Elasticsearch::Model` changed.